Add -k flag and https to 2 examples

mellybess opened the pull request:
In a recent customer question (in Front, so no thread available :frowning:) @davideg @brainstarr and I found that the example commands were missing a `-k` flag, and needed to be via https. The steps we documented for the customer are [here](https://gist.github.com/brainstarr/03aabb95bd4154ea3da6).

I'm not sure if this needs to be updated for each example on the page, but I can test that if needed.

I also removed a version number from a link to the Enterprise docs so that it redirects to the latest version.

/cc @github/api, @github/docs-platform, and @github/copy

pengwynn commented on content/v3/enterprise/management_console.md:
````diff
@@ -69,7 +69,7 @@ For a list of the available settings, see [the `/setup/api/settings` endpoint](#
 
 ``` command-line
-$ curl -L -X POST 'http://<em>hostname</em>:<em>admin_port</em>/setup/api/start' -F license=@<em>/path/to/github-enterprise.ghl</em> -F "password=<em>your-amazing-password</em>" -F settings=&lt;<em>/path/to/settings.json</em>
+$ curl -L -X -k POST 'https://<em>hostname</em>:<em>admin_port</em>/setup/api/start' -F license=@<em>/path/to/github-enterprise.ghl</em> -F "password=<em>your-amazing-password</em>" -F settings=&lt;<em>/path/to/settings.json</em>
````
I think the flag order might matter here since `POST` is an argument to `-X`. Should this be `-X POST -k ...`?

jacobbednarz commented on content/v3/enterprise/management_console.md:
````diff
@@ -69,7 +69,7 @@ For a list of the available settings, see [the `/setup/api/settings` endpoint](#
 
 ``` command-line
-$ curl -L -X POST 'http://<em>hostname</em>:<em>admin_port</em>/setup/api/start' -F license=@<em>/path/to/github-enterprise.ghl</em> -F "password=<em>your-amazing-password</em>" -F settings=&lt;<em>/path/to/settings.json</em>
+$ curl -L -X POST -k 'https://<em>hostname</em>:<em>admin_port</em>/setup/api/start' -F license=@<em>/path/to/github-enterprise.ghl</em> -F "password=<em>your-amazing-password</em>" -F settings=&lt;<em>/path/to/settings.json</em>
````
Is allowing insecure connections something you'd want to promote via documentation? Passing in the `-k` flag will ignore any SSL warnings (such as mismatching certificates) and allow the connection to proceed anyway which has some potential security concerns. Personally, if I was sending my GitHub license or credentials across the wire I'd definitely like to ensure the host I am connecting to is who I intend it to be. However, on the flip side of that, I do realise a lot of places that setup on their internal network won't bother with valid certificates so `¯\_(ツ)_/¯`.

gjtorikian replied:
@jacobbednarz GitHub Enterprise provides a self-signed certificate. For many, I assume the lack of `-k` blocks these admins from using `curl` to work with the API.

jacobbednarz replied:
:+1:

dbussink replied:
Enterprise customers should configure a proper certificate though and that is strongly recommended. I would be in favor of not adding `-k` and only add a comment for using it if absolutely necessary. Always adding as a blanket solution is not a good idea.

davideg replied:
:+1: I think adding a comment explaining the `-k` option is the right approach.

gjtorikian commented on content/v3/enterprise/management_console.md:
```diff
@@ -18,7 +18,7 @@ If you don't want to provide a port number, you'll need to configure your tool t
 ## Authentication
 
-You need to pass your [Management Console password](https://help.github.com/enterprise/2.0/admin/articles/accessing-the-management-console/) as an authentication token to every Management Console API endpoint except [`/setup/api/start`](#upload-a-license-for-the-first-time).
+You need to pass your [Management Console password](https://help.github.com/enterprise/admin/articles/accessing-the-management-console/) as an authentication token to every Management Console API endpoint except [`/setup/api/start`](#upload-a-license-for-the-first-time).
```
Great catch!

gjtorikian commented:
> I'm not sure if this needs to be updated for each example on the page, but I can test that if needed.

I just tested it; looks like `curl` to HTTP will *not* work. I used [the maintenance settings check](https://developer.github.com/v3/enterprise/management_console/#check-maintenance-status) as it's one of the few `GET` requests on there.

I did *not* pass `-k`, but I don't really know the intricacies of the cert dance. It's interesting to note also that this was pointed out to [need to be HTTPS](https://github.com/github/developer.github.com/pull/859#issuecomment-142391624) but we didn't change it. :sweat: 

If you could change the other examples to `https` that would be great! I am not sure if they need `-k` or not but verifying that would be cool. :+1:

kosikfll commented:
Do you have the host name I know I wanted to change it .  Maybe we should all put this to a bow on what it should be I deathly want my name in it because that's what the UK wanted so I want to .

kosikfll commented:
Need to register as it  as a global company but do you all want to be in it too .

kosikfll commented:
I need a new organization do you want to be my organization ?

kosikfll commented:
We need to talk you're basically the backers financially find the name and the history and the story what do you say  65% my end 35%  your half  do you think this is fair do you want to bring it to lawyers ?

kosikfll commented:
I have to pay royalties for music and other finances for royalties that's why my percentage is higher .

kosikfll commented:
We need to decide on a name before I upload it do you want to be the hot hidden back her because then marketing name  name is very  important ?

kosikfll commented:
Michelle, Lee Kosik,   Tiger Hsck & Company

kosikfll commented:
Michelle Ler Kodik   Tiger hack & Company

kosikfll commented:
Get back to me in the multiple of this OK.  Get back to me in the multiple of this OK It's important I have my name my name's and all the books and I haven't had it in so long

davideg commented on content/v3/enterprise/management_console.md:
```diff
@@ -47,6 +47,8 @@ When using this endpoint, your Enterprise instance must have a password set. Thi
 1. If you're working directly with the API before accessing the web interface, you must pass in the password parameter to set your password.
 2. If you set up your instance via the web interface before accessing the API, your calls to this endpoint do not need the password parameter.
 
+You may also need to use the `-k` flag, as GitHub Enterprise uses a self-signed certificate before you [enable your own SSL certificate](https://help.github.com/enterprise/admin/guides/installation/dns-hostname-subdomain-isolation-and-ssl/#ssl).
```
I would make it clear that the `-k` flag applies to using curl.

mellybess commented:
I've added a note about needing the -k flag and updated http to https in the examples.

@pengwynn @github/docs-platform @github/copy is this :+1:  to merge?

davideg commented on content/v3/enterprise/management_console.md:
```diff
@@ -47,6 +47,8 @@ When using this endpoint, your Enterprise instance must have a password set. Thi
 1. If you're working directly with the API before accessing the web interface, you must pass in the password parameter to set your password.
 2. If you set up your instance via the web interface before accessing the API, your calls to this endpoint do not need the password parameter.
 
+You may also need to add the `-k` flag when using `curl`, since GitHub Enterprise uses a self-signed certificate before you [add your own SSL certificate](https://help.github.com/enterprise/admin/guides/installation/dns-hostname-subdomain-isolation-and-ssl/#ssl).
```
@mellybess I would add a small bit about what the `-k` flag does or at least add a link to http://curl.haxx.se/docs/manpage.html#-k (not sure if linking out to external pages in these docs is cool or not).

Also since this tip applies when using any self-signed certificate it could apply to all endpoints. I'm wondering if it would make sense to move it up towards the top of the document after the tip section.
